### PR TITLE
chore: remove unused variables

### PR DESCRIPTION
**Pull request by chandumlg**
# Description

1. Clean up stale variables
2. Use proper duration for alerting

## Notion Ticket

https://www.notion.so/rudderstacks/8aac9087df644365acdf64e28e290153?v=6e06b0a5ade24f0aa5ffe05dc2972e84&p=d83fc81b3a284659a0c40875810c10f8

## Security

- [x] The code changed/added as part of this pull request won't create any security issues with how the software is being used.

### CODE REVIEW

File: router/router.go [hunk 1]
```diff
@@ -2211,6 +2201,7 @@ func (rt *HandleT) Setup(backendConfig backendconfig.BackendConfig, jobsDB jobsd
 	rt.destName = destName
 	netClientTimeoutKeys := []string{"Router." + rt.destName + "." + "httpTimeout", "Router." + rt.destName + "." + "httpTimeoutInS", "Router." + "httpTimeout", "Router." + "httpTimeoutInS"}
 	config.RegisterDurationConfigVariable(10, &rt.netClientTimeout, false, time.Second, netClientTimeoutKeys...)
+	config.RegisterDurationConfigVariable(30, &rt.timeoutDuration, false, time.Second, []string{"HttpClient.timeout"}...)
```
[the remark this answers is not in the document]
Comment (chandumlg) [reply router/router.go:2204]: Yes. `HttpClient.timeout` is already used by other http clients (ex: http client in router transformer)

**Review by cisse21 — APPROVED** [review]

File: router/router.go [hunk 2]
```diff
@@ -103,6 +102,7 @@ type HandleT struct {
 	throttler                              throttler.Throttler
 	guaranteeUserEventOrder                bool
 	netClientTimeout                       time.Duration
+	timeoutDuration                        time.Duration
```
[the remark this answers is not in the document]
Comment (chandumlg) [reply router/router.go]: done

File: router/router.go [hunk 3]
```diff
@@ -890,7 +883,7 @@ func (worker *workerT) processDestinationJobs() {
 				// This is introduced to ensure the above statement
 				resp := misc.UpdateJSONWithNewKeyVal(router_utils.EmptyPayload, "blocking_id", prevFailedJobID)
 				resp = misc.UpdateJSONWithNewKeyVal(resp, "user_id", destinationJobMetadata.UserID)
-				resp = misc.UpdateJSONWithNewKeyVal(resp, "moreinfo", "attempted to send in a batch")
+				resp = misc.UpdateJSONWithNewKeyVal(resp, "more info", "attempted to send in a batch")
```
[the remark this answers is not in the document]
Comment (chandumlg) [reply router/router.go:886]: 😄  guessed it

File: router/router.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (atzoum) [reply router/router.go]: what is the rationale behind binding the alert's min timeout limit with this http timeout?

[the remark this answers is not in the document]
Comment (chandumlg) [reply router/router.go]: We fire delivery timeout alert when the delivery instructions take more than 2*netClientTimeout (10 seconds default).

But if transformer proxy is enabled, this code can take longer and that is expected. So, we are seeing false alerts.

_______Start timer________

if transformerProxy enabled
      make a http call to transformer <---------- timeout 30 seconds
else if customerDestinationManager
       sendData <----------- timeout 10 seconds
else
       http post call <----------- timeout 10 seconds

_______End timer_________
            
If the timer exceeds 20 seconds, we get an alert currently.

So, in this pr, we are using max of both (10, 30) to start the timer in alerting goroutine.

[the remark this answers is not in the document]
Comment (chandumlg) [reply router/router.go]: Makes sense :)

Thought it would demand a lot of changes. Actually not.

Made the changes.

File: utils/misc/misc.go [hunk 4]
```diff
@@ -1420,3 +1420,11 @@ func CopyStringMap(originalMap map[string]string) map[string]string {
 	}
 	return newMap
 }
+
+func MaxDuration(a, b time.Duration) time.Duration {
```
Comment (atzoum) [inline utils/misc/misc.go:1424]: is this used?

File: router/router.go [hunk 5]
```diff
@@ -103,6 +102,7 @@ type HandleT struct {
 	throttler                              throttler.Throttler
 	guaranteeUserEventOrder                bool
 	netClientTimeout                       time.Duration
+	minDeliveryTimeoutAlertLimit           time.Duration
```
Comment (atzoum) [inline router/router.go:105]: backendProxyTimeout might be a better name after the most recent changes

File: router/router.go [hunk 6]
```diff
@@ -577,6 +570,17 @@ func (worker *workerT) workerProcess() {
 	}
 }
 
+func (worker *workerT) getDeliveryTimeout() time.Duration {
```
Comment (atzoum) [inline router/router.go:573]: this can be inlined inside trackStuckDelivery

File: router/router.go (hunk 5 above)
Comment (chandumlg) [reply router/router.go]: done

**Review by cisse21 — APPROVED** [review]

**Review by atzoum — APPROVED** [review]

**Pull request merged**